File: medplum-java/medplum-model/src/main/java/com/fhir/ObservationDefinition_QualifiedInterval.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableObservationDefinition_QualifiedInterval.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface ObservationDefinition_QualifiedInterval {
  Optional<Range> range();

  Optional<String> condition();

  Optional<List<Extension>> modifierExtension();

  Optional<Observationdefinition_qualifiedintervalGender> gender();

  Optional<List<Extension>> extension();

  Optional<Range> gestationalAge();

  Optional<Observationdefinition_qualifiedintervalCategory> category();

  Optional<List<CodeableConcept>> appliesTo();

  Optional<CodeableConcept> context();

  Optional<Range> age();

  Optional<String> id();

  static ImmutableObservationDefinition_QualifiedInterval.Builder builder() {
    return ImmutableObservationDefinition_QualifiedInterval.builder();
  }
}

File: medplum-typescript/src/model/fhir/ObservationDefinition_QualifiedInterval.ts
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { Observationdefinition_qualifiedintervalCategory } from "./Observationdefinition_qualifiedintervalCategory";
import { Observationdefinition_qualifiedintervalGender } from "./Observationdefinition_qualifiedintervalGender";
import { Range } from "./Range";

export interface ObservationDefinition_QualifiedInterval {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    category:
        | Observationdefinition_qualifiedintervalCategory
        | null
        | undefined;
    range: Range | null | undefined;
    context: CodeableConcept | null | undefined;
    appliesTo: CodeableConcept[] | null | undefined;
    gender: Observationdefinition_qualifiedintervalGender | null | undefined;
    age: Range | null | undefined;
    gestationalAge: Range | null | undefined;
    condition: string | null | undefined;
}
